metaseq/data/streaming_src_tgt_dataset.py

```diff
@@ -207,3 +215,49 @@ def yield_src_tgt_single_sentences_pad_8(iterable, block_size, drop_last, paddin
             "src_block": torch.cat(cur_src_block),
             "tgt_block": torch.cat(cur_tgt_block),
         }
+
+
+def yield_src_tgt_single_sentences_withneg_pad_8(
+    iterable, block_size, drop_last, padding_idx
+):
+    """
+    Mimics sample-break-mode eos_pad_8 with additionally adding negative candidates.
+    """
+    counter = 0
+    for all_src, all_tgt in iterable:
+        if not isinstance(all_src, tuple):
+            all_src = (all_src,)
+            all_tgt = (all_tgt,)
+        for idx, (src, tgt) in enumerate(zip(all_src, all_tgt)):
+            cur_src_block = []
+            cur_src_block_ids = []
+            cur_tgt_block = []
+            if src.numel() > block_size:
+                # truncate right side
+                # TODO: Enable left side truncation
```

Comment:
Switch to left-truncation now?

Reply:
this module is removed, but it's a good point. Let's decide on this truncation method and update it in a separate PR.